### PR TITLE
Signup

### PR DESCRIPTION
**Pull request by TeriPastorino**
@levicole 
hey levi added buttons and trying to play with the formatting. can you tell me how i get containers and rows in the proper syntax to get buttons off of each other? will work on session controller tomorrow

### CODE REVIEW

**Comment by levicole** [comment]
so, your application.css need to be rename applicaiton.scss, and the things you are importing aren't correct.

When I get all that fixed, this is what the page looks like:

https://www.dropbox.com/s/sq2yhmf0qjkm4qy/Screenshot%202015-05-10%2010.37.40.png?dl=0

``` scss
@import "bootstrap-sprockets";
@import "bootstrap";
@import "font-awesome";
@import "bootstrap-social";
```

That is what I have in the application.scss.

Also, a few other things. In general, when you are using scss/sass, you don't need to touch the requires in the top comments. If you are going to add a new gem that contains assets, you just have to add it to the imports. If they tell you to require it, just take the part to the right of the require and add that as an import instead :).

I noticed you had multiple buttons on the page, let's try to just get one button working first before we try adding more than one. The mark up is pretty close. Try something like this:

``` haml
%a.btn.btn-social.btn-lg.btn-twitter{:href => "/auth/twitter", :role => "button"}
  %i.fa.fa-twitter &nbsp;
  Sign in with Twitter
```

Don't worry about converting that link to a link_to helper, it's actually easier, I think, to do it without in this case.

**Comment by levicole** [comment]
Notice, I removed .btn-block class. That will make it so that it's a set size, and not the entire width of it's container. Now, with that said, you will need to wrap this link in a block level element like a `p` or a `div`

**Comment by levicole** [comment]
I love seeing so many commits.

File: spec/controllers/welcome_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
+RSpec.describe WelcomeController, type: :controller do
+
+  describe "GET #index" do
+    it "returns http success" do
```
Comment (levicole) [inline spec/controllers/welcome_controller_spec.rb:6]: I like this!

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class SessionsController < ApplicationController
+  def create
+    user = User.find_or_create_by_omniauth(auth_hash)
+    session[:uid] = @uid
```
Comment (levicole) [inline app/controllers/sessions_controller.rb]: Where is @uid coming from?

Also, the key in the session has to match what you are using in our current_user method. What attribute are we finding by in the current_user method in application controller.

**Comment by TeriPastorino** [comment]
is a user_id set up as a default when you generate a model? because i don't have one in my db
 i had user_id originally and was getting some type of login error. so looked in schema and no user_id just uid. 
thought it was tied to that. changed it back to user_id

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -2,4 +2,14 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  helper_method :current_user
+
+  def current_user
+    #look up once
+    @current_user ||= User.find(session[:uid]) if session[:uid]
```
Comment (levicole) [inline app/controllers/application_controller.rb]: This method was actually ok before. The only thing that needed to change was line 4 of SessionsController. The key in the session here, should match the key in the session on line 4 in the SessionsController.

Comment (TeriPastorino) [inline app/controllers/application_controller.rb]: fixed and committed 

----- Original Message -----

**Pull request merged**